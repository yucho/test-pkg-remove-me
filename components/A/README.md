# @test-pkg-remove-me/a

Welcome to my `A` component.

## Installation

```sh
npm i @test-pkg-remove-me/a
# or with yarn
yarn add @test-pkg-remove-me/a
```

Then to use the component in your code just import it!

```js
import A from '@test-pkg-remove-me/a';
// and with css-modules
import '@test-pkg-remove-me/a/dist/main.css';
```
