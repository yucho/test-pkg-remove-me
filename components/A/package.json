{
  "name": "@test-pkg-remove-me/a",
  "version": "0.0.3-next.0",
  "main": "./dist/cjs/index.js",
  "module": "./dist/esm/index.js",
  "types": "./dist",
  "repository": "https://github.com/yucho/test-pkg-remove-me.git",
  "author": "Yucho Ho yucho.ho@gmail.com",
  "contributors": [
    "Yucho Ho yucho.ho@gmail.com"
  ],
  "license": "MIT",
  "sideEffects": [
    "*.css"
  ],
  "scripts": {
    "clean": "ds clean",
    "build": "ds build",
    "dev": "ds dev",
    "storybook": "ds storybook start",
    "start": "ds build --watch",
    "test": "ds test",
    "lint": "ds lint",
    "size": "ds size"
  },
  "dependencies": {
    "@babel/runtime": "7.11.2"
  },
  "peerDependencies": {
    "react": ">= 16.8.6"
  },
  "files": [
    "dist",
    "src",
    "!*.snippet.*",
    "!*theme.*",
    "!*.test.*",
    "!__snapshots__",
    "!__tests__"
  ]
}
