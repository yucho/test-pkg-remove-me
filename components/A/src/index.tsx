import React from 'react';
import styles from './A.css';

interface AProps {
  /** The value for the component to display */
  value?: string;
}

/** A simple component. */
export const A = (props: AProps) => {
  return (
    <>
      <div className={styles.a}>This is a A component {props.value}</div>
      <div>Add line to increase component size</div>
    </>
  );
};

export default A;
