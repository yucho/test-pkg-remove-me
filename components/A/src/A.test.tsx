import React from 'react';
import { render } from '@testing-library/react';
import A from '.';

describe('A', () => {
  test('It matches the snapshot', () => {
    const { container } = render(<A />);
    expect(container).toMatchSnapshot();
  });
});
