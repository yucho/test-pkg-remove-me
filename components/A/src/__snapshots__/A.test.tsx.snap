// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A It matches the snapshot 1`] = `
<div>
  <div
    class="a"
  >
    This is a A component 
  </div>
</div>
`;
